### ISSUE TITLE
anyhow not working with `async fn`

### ISSUE THREAD

**Issue by probablykasper** [napi-rs/napi-rs#2278]
I'm getting the following error when using `#[napi]` on an `async fn` that returns an `anyhow::Result`:

```rs
error[E0271]: expected `{async block@src-native/tracks/cover.rs:117:1: 117:8}` to be a future that resolves to `Result<_, Error>`, but it resolves to `Result<Option<Buffer>, Error>`
   --> src-native/tracks/cover.rs:117:1
    |
117 | #[napi]
    | ^^^^^^^ expected `napi::Error`, found `anyhow::Error`
    |
    = note: `anyhow::Error` and `napi::Error` have similar names, but are actually distinct types
note: `anyhow::Error` is defined in crate `anyhow`
   --> /Users/k/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.89/src/lib.rs:389:1
    |
389 | pub struct Error {
    | ^^^^^^^^^^^^^^^^
note: `napi::Error` is defined in crate `napi`
   --> /Users/k/.cargo/registry/src/index.crates.io-6f17d22bba15001f/napi-2.16.10/src/error.rs:24:1
    |
24  | pub struct Error<S: AsRef<str> = Status> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: required by a bound in `execute_tokio_future`
   --> /Users/k/.cargo/registry/src/index.crates.io-6f17d22bba15001f/napi-2.16.10/src/tokio_runtime.rs:153:32
    |
151 | pub fn execute_tokio_future<
    |        -------------------- required by a bound in this function
152 |   Data: 'static + Send,
153 |   Fut: 'static + Send + Future<Output = Result<Data>>,
    |                                ^^^^^^^^^^^^^^^^^^^^^ required by this bound in `execute_tokio_future`
    = note: this error originates in the attribute macro `napi` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0271`.
error: could not compile `ferrum` (lib) due to 1 previous error
```

**Comment by richerfu** [comment]
I think you should upgrade napi crate version.
Related PR: https://github.com/napi-rs/napi-rs/pull/2165/files
Related issue: https://github.com/napi-rs/napi-rs/issues/2178

**Comment by probablykasper** [comment]
That's only available in the 3.0.0 alpha, right?

I'll stick to stable, but I can try upgrading to test it out. Is there any guide for the breaking changes? Facing these errors:

https://github.com/probablykasper/ferrum/actions/runs/11109657314/job/30865521961

> ```rs
> 582 | #[napi]
>     | ^^^^^^^
>     | |
>     | private function
>     | function `raw_finalize_unchecked` is not publicly re-exported
> ```
> 
> ```rs
> 582 | #[napi]
>     | ^^^^^^^
>     | |
>     | expected `TypeId`, found `&str`
>     | arguments to this function are incorrect
> ```

**Comment by probablykasper** [comment]
Ah, sorry. It works!
